src/pocketmine/entity/Entity.php

```diff
@@ -907,6 +908,19 @@ public function attack(EntityDamageEvent $source) : void{
 		$this->setHealth($this->getHealth() - $source->getFinalDamage());
 	}
 
+	/**
+	 * Called when a player uses the item on an entity, for example shearing a sheep.
+	 * Returns whether the item was changed, for example count decrease or durability change.
+	 *
+	 * @param Player  $player
+	 * @param Item    $heldItem
+	 * @param Vector3 $clickPos
+	 * @return bool
+	 */
+	public function onPlayerInteract(Player $player, Item $heldItem, Vector3 $clickPos) : bool {
```

Comment:
I think change the function's name to ``onInteract`` and You must change ``: bool {`` to ``: bool{``